token/wallet.tact

```diff
@@ -2,164 +2,116 @@
         })
-        }.serialize())
-        .send_to(to_wallet_address);
-}
-
-fn receive_tokens(
-    @mut state: WalletState, 
-    msg: InternalTransfer, 
-    sender: ContractAddress, 
-    my_ton_balance: Coins, 
-    fwd_fee: Coins, 
-    msg_value: Coins
-) {
-    verify(
-        sender === state.master || 
-            sender == WalletAddress.from_data(msg.from, state.master, state.wallet_code), 
-        "sender must be master or user wallet"
-    );
-
-    let ton_balance_before_msg = my_ton_balance - msg_value;
-    let storage_fee = STORAGE_MIN_CONSUMPTION - ton_balance_before_msg.min(STORAGE_MIN_CONSUMPTION);
-    let @mut msg_value = msg_value - (storage_fee + GAS_CONSUMPTION);
-
-    if (forward_ton_amount != 0) {
-        msg_value -= (forward_ton_amount + fwd_fee);
-        let payload = msg.forward_payload;
-        RequestBuilder.new()
-            .ihr_disabled()
-            .pay_transfer_fee_separately()
-            .money(forward_ton_amount)
-            .body(TransferNotification {
-                query_id: msg.query_id,
-                amount: msg.amount,
-                from: msg.from,
-                payload: payload,
-            })
-            .send_to(owner_address);
+        .send(
+            ihr_disabled: true,
+            can_be_bounced: true,
+            carry_message_input: true,
+        );
     }
 
-    if (!response_address.is_null() && msg_value > 0) {
-        RequestBuilder.new()
-            .ihr_disabled()
-            .ignore_action_phase_errors()
-            .money(msg_value)
-            .body(Excesses {
-                query_id: msg.query_id,
-            })
-            .send_to(response_address);
+    internal receive_tokens#178d4519(transfer: InternalTransfer) {
+        verify(
+            msg.sender === self.master || 
+                sender == WalletAddress.from_data(msg.from, self.master, self.wallet_code), 
+            "sender must be master or user wallet"
+        );
+
+        let ton_balance_before_msg = self.my_balance() - msg.amount;
+        let storage_fee = STORAGE_MIN_CONSUMPTION - ton_balance_before_msg.min(STORAGE_MIN_CONSUMPTION);
+        let mut msg_value = msg.amount - (storage_fee + GAS_CONSUMPTION);
+
+        if (transfer.forward_ton_amount != 0) {
+            msg_value -= (transfer.forward_ton_amount + fwd_fee);
+            let payload = msg.forward_payload;
+            self.owner
+                .notify_transfer(
+                    transfer.query_id,
+                    transfer.amount,
+                    msg.from,
+                    payload,
+                )
+                .send(ihr_disabled: true, pay_transfer_fee_separately: true);
+        }
+
+        if (!transfer.response_address.is_none() && msg_value > 0) {
+            transfer
+                .response_address
+                .excesses(msg.query_id)
+                .send(
+                    money: msg.amount,
+                    ihr_disabled: true,
```

Comment:
we should use "ihr_disabled: true" as default since IHR does not work yet. When it will, we'll either change the default or user would opt-in explicitly with "ihr: true" or something.